codebleu/test_codebleu.py

```diff
@@ -0,0 +1,180 @@
+import json
+import unittest
+import collections
+import codebleu_metric as cbl
+import codebleu.graph_generator.graphgenerator as gg
+import networkx as nx
+from graph_generator.graphgenerator import AstGraphGenerator
+from graph_generator.graphgenutils import prettyprint_graph
+from graph_generator.type_lattice_generator import TypeLatticeGenerator
+from tree_sitter import Language, Parser
+
+
+class TestCodebleuMetric(unittest.TestCase):
+    # ---- DFG part ----
+    parser = Parser()
+    PY_LANGUAGE = Language('codebleu/my-languages.so', 'python')
+    parser.set_language(PY_LANGUAGE)
+    lattice = TypeLatticeGenerator('codebleu/typingRules.json')
+
+    def test_inlist(self):
+        self.assertEqual(cbl._inlist(1, [1, 2, 3]), True)
+        self.assertEqual(cbl._inlist(1, [5, 2, 3]), False)
+        self.assertEqual(cbl._inlist(1, 1), True)
+        self.assertEqual(cbl._inlist(1, 2), False)
```

Comment:
The downside of using 4 separate assert statements, in this case, is that if the first assert fails, you won't see the test results for the next ones. It is better to use [parameterized tests](https://docs.pytest.org/en/latest/how-to/parametrize.html) here.

```suggestion
    @pytest.mark.parametrize(
        "value, lst, contained",
        [(1, [1, 2, 3], True), (1, [5, 2, 3], False), (1, 1, True), (1, 2, False)],
    )
    def test_inlist(self, value, lst, contained):
        assert cbl._inlist(value, lst) == contained
```

Note the usage of the plain `assert` statement as we do not inherit from `unittest.TestCase` anymore.
This way you will see four different tests when running just a single one:
```
codebleu/test_codebleu.py::TestCodebleuMetric::test_inlist[1-lst0-True] PASSED [ 25%]
codebleu/test_codebleu.py::TestCodebleuMetric::test_inlist[1-lst1-False] PASSED [ 50%]
codebleu/test_codebleu.py::TestCodebleuMetric::test_inlist[1-1-True] PASSED [ 75%]
codebleu/test_codebleu.py::TestCodebleuMetric::test_inlist[1-2-False] PASSED [100%]
```